paymentsheet/src/main/java/com/stripe/android/paymentsheet/ExternalPaymentMethodActivity.kt

```diff
@@ -0,0 +1,65 @@
+package com.stripe.android.paymentsheet
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * Handles external payment methods.
+ *
+ * This will call the merchant's [ExternalPaymentMethodConfirmHandler.confirmExternalPaymentMethod] when initially
+ * created. When the merchant's implementation calls [ExternalPaymentMethodResultHandler.onExternalPaymentMethodResult],
+ * it will re-start this activity with the result. This class then finishes and the class that launched this activity
+ * is responsible for responding to that result.
+ */
+internal class ExternalPaymentMethodActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val type = intent.getStringExtra(EXTRA_EXTERNAL_PAYMENT_METHOD_TYPE)
+
+        @Suppress("DEPRECATION")
+        val billingDetails = intent.getParcelableExtra<PaymentMethod.BillingDetails>(EXTRA_BILLING_DETAILS)
+
+        if (type != null) {
```

Comment:
Should we `finish` if it is null?

Reply:
No, if the activity is re-created by the `onExternalPaymentMethodResult` call, then this function would be called without an external payment method type. We'd want the activity to keep running until `onResume`, at which point the external payment method result would be parsed and _then_ the activity would finish.

Reply:
Hmm, I'm thinking of the case that the merchant launches this, without passing in any arguments.

Or an attacker is testing activities, and launches this via Instrumentation. 

In those cases, I'd think we'd just want to finish the activity without any actions.

All that being said, I don't think it's super important, given this activity does almost nothing, and we don't need any state for it to run.

Reply:
ah ok, good to know, I'll think about if we can handle those cases